### PR TITLE
update to support asynchronous callbacks from the browser

### LINKED ISSUE #360: Hard to listen to all-test-results event
The code of testem client is

```js
window.Testem = {
  emit: function(evt){
    var args = Array.prototype.slice.apply(arguments)
    socket.emit.apply(socket, circularSafe(args))
    if (this.evtHandlers && this.evtHandlers[evt]){
      var handlers = this.evtHandlers[evt]
      var args = Array.prototype.slice.call(arguments, 1)
      for (var i = 0; i < handlers.length; i++){
        var handler = handlers[i]
        handler.apply(this, args)
      }
    }
  },
```

Which means, first the socket then the callbacks. In case of `all-test-results`, the socket message tells `testem ci` to kill the browser, so depending on how quick this happen, the handler callback might not be called.

It would be good to move the socket.emit at the end of the method

### PR DESCRIPTION
**Pull request by jreading**
First pass at having asynchronous callbacks from the browser. 
Fixes #360

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -409,6 +409,34 @@ If you need to run a command after your tests have completed (such as removing c
...
 If you would prefer simply to clean up when Testem exits, you can use the `on_exit` option.
 
+Running browser code after tests complete
+-------------
+It is possible to send coverage reports or run other javascript in the browser by using the `afterTests` method. For asynchronous functions, just run a callback function as the first argument.
+
+```javascript
+Testem.afterTests(
+  //Asynchronously
+  function(cb) {
```
Comment (johanneswuerbach) [inline README.md]: I would prefer if we use the same API as you can already use in the config `function(config, data, callback)` to allow passing more additional details in the future.

https://github.com/testem/testem/blob/784e210f0619fcebd32f5ac51062396597a38e5f/examples/dynamic_config/testem.js#L16

I would also drop the `{"async": true}` and require the callback by default.

**Comment by jreading** [comment]
This should be ready now.

**Comment by johanneswuerbach** [comment]
Nearly, could you also update this example https://github.com/testem/testem/tree/master/examples/coverage_istanbul to use your technic? Examples are continuously tested, so this should improve the test coverage here a bit.

Otherwise, LGTM :+1:

**Comment by johanneswuerbach** [comment]
Thanks :+1: could you squash your commits then I can merge!

**Comment by johanneswuerbach** [comment]
Looks like the test failures are related, could you have a look?

**Comment by jreading** [comment]
tap runner fixed.

**Pull request merged**